lib.py

```diff
@@ -65,41 +69,74 @@ def parse_checkmarks(body):
     return commits
 
 
-def get_release_pr(org, repo, version):
+def _get_pr(org, repo, branch):
     """
-    Look up the release pull request
+    Look up the pull request for a branch
 
     Args:
         org (str): The github organization (eg mitodl)
         repo (str): The github repository (eg micromasters)
-        version (str): A version string used to match the PR title
+        branch (str): The name of the associated branch
 
     Returns:
-        dict: The information about the release pull request
+        dict: The information about the pull request
     """
-    pulls = requests.get("https://api.github.com/repos/{org}/{repo}/pulls".format(
+    response = requests.get("https://api.github.com/repos/{org}/{repo}/pulls".format(
         org=org,
         repo=repo,
-    )).json()
-    release_pulls = [pull for pull in pulls if pull['title'] == "Release {}".format(version)]
-    if len(release_pulls) == 0:
-        raise ReleaseException("No release pull request on server")
-    elif len(release_pulls) > 1:
-        raise ReleaseException("More than one release pull request open at the same time")
+    ))
+    response.raise_for_status()
+    pulls = response.json()
+    pulls = [pull for pull in pulls if pull['head']['ref'] == branch]
+    if len(pulls) == 0:
+        return None
+    elif len(pulls) > 1:
+        # Shouldn't happen since we look up by branch
+        raise Exception("More than one pull request for the branch {}".format(branch))
+
+    return pulls[0]
+
+
+def get_release_pr(org, repo):
+    """
+    Look up the pull request information for a release, or return None if it doesn't exist
+
+    Args:
+        org (str): The github organization (eg mitodl)
+        repo (str): The github repository (eg micromasters)
+
+    Returns:
+        ReleasePR: The information about the release pull request, or None if there is no release PR in progress
+    """
+    pr = _get_pr(org, repo, 'release-candidate')
+    if pr is None:
+        return None
+
+    title = pr['title']
+    match = re.match(r'^Release (?P<version>\d+\.\d+\.\d+)$', title)
+    if not match:
+        raise ReleaseException("Release PR title has an unexpected format")
+    version = match.group('version')
 
-    return release_pulls[0]
+    return ReleasePR(
+        version=version,
+        body=pr['body'],
+        url=pr['html_url'],
+    )
 
 
-def get_unchecked_authors(org, repo, version):
+def get_unchecked_authors(org, repo):
     """
     Returns list of authors who have not yet checked off their checkboxes
 
     Args:
         org (str): The github organization (eg mitodl)
         repo (str): The github repository (eg micromasters)
-        version (str): A version string used to match the PR title
     """
-    body = get_release_pr(org, repo, version)['body']
+    release_pr = get_release_pr(org, repo)
+    if not release_pr:
+        raise ReleaseException("No release PR found")
+    body = release_pr['body']
```

Comment:
Nit: better to use the dot notation here

Reply:
Yeah, this is actually a bug but the tests didn't catch it because I didn't change the mocked return value of the function. I'll do that now